### PR TITLE
build: tidy dockerfile

### PR DESCRIPTION
**Pull request by tisonkun**
Comments inline.

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -1,5 +1,54 @@
...
+version.h
+
+.idea
+.vscode
+.cache
+
+compactdb
+testdb
+
+build
```
Comment (tisonkun) [inline .dockerignore:53]: Add `.gitignore` content. Avoid build content conflict issues.

File: Dockerfile [hunk 2]
```diff
@@ -19,10 +19,6 @@ FROM alpine:3.16 as build
 
 ARG MORE_BUILD_ARGS
 
-# workaround tzdata install hanging
-ENV TZ=Etc/UTC
```
Comment (tisonkun) [inline Dockerfile:23]: After switching to Alpine, we don't need this trick.

File: Dockerfile [hunk 3]
```diff
@@ -31,36 +27,23 @@ RUN ./x.py build -DENABLE_OPENSSL=ON -DPORTABLE=ON -DCMAKE_BUILD_TYPE=Release -j
...
-COPY --from=build /usr/bin/redis-cli ./bin/
-
-HEALTHCHECK --interval=10s --timeout=1s --start-period=30s --retries=3 CMD ./bin/redis-cli -p 6666 PING | grep -E '(PONG|NOAUTH)' || exit 1
-
-ENV PATH="$PATH:/kvrocks/bin"
-
 VOLUME /var/lib/kvrocks
 
-RUN chown kvrocks:kvrocks /var/lib/kvrocks
+COPY --from=build /kvrocks/build/kvrocks /bin/
```
Comment (tisonkun) [inline Dockerfile:38]: Simplify a bit layout:

```
/bin/kvrocks
/bin/redis-cli
/kvrocks/<LICENSE/NOTICE>
/var/...
```

File: Dockerfile [hunk 4]
```diff
@@ -31,36 +27,23 @@ RUN ./x.py build -DENABLE_OPENSSL=ON -DPORTABLE=ON -DCMAKE_BUILD_TYPE=Release -j
...
-COPY --from=build /kvrocks/build/kvrocks ./bin/
-COPY --from=build /usr/bin/redis-cli ./bin/
-
-HEALTHCHECK --interval=10s --timeout=1s --start-period=30s --retries=3 CMD ./bin/redis-cli -p 6666 PING | grep -E '(PONG|NOAUTH)' || exit 1
-
-ENV PATH="$PATH:/kvrocks/bin"
-
 VOLUME /var/lib/kvrocks
 
-RUN chown kvrocks:kvrocks /var/lib/kvrocks
```
Comment (tisonkun) [inline Dockerfile:58]: Duplicated above.

**Comment by tisonkun** [comment]
cc @aleksraiden may you verify this patch also.

**Comment by aleksraiden** [comment]
> cc @aleksraiden may you verify this patch also.

Awesome! LGTM

**Review by torwig — APPROVED** [review]

**Comment by tisonkun** [comment]
Merging...

Thanks for your review!

**Pull request merged**